packages/react-shopify-app-bridge/src/Provider.tsx

```diff
@@ -266,17 +280,31 @@ type ProviderLocation = {
   asPath?: string;
 };
 
-export const Provider = ({ type, children, api }: { type?: AppType; children: ReactNode; shopifyApiKey: string; api: AnyClient }) => {
+export const Provider = ({
+  type,
+  children,
+  api,
+  location: _location,
+  shopifyInstallState,
+}: {
+  type?: AppType;
+  children: ReactNode;
+  shopifyApiKey: string;
+  api: AnyClient;
+  location?: Pick<Window["location"], "pathname" | "search">;
+  shopifyInstallState?: ShopifyInstallState;
+}) => {
   // if we haven't properly set up the shopify global then skip anything that requires app bridge
   const shopifyGlobalDefined = !!(globalThis && globalThis.shopify);
 
-  const location = useMemo<ProviderLocation>(
-    () => ({
-      asPath: `${window.location.pathname}${window.location.search}`,
-      query: new URLSearchParams(window.location.search),
-    }),
-    []
-  );
+  const location = useMemo<ProviderLocation>(() => {
+    const pathname = _location?.pathname ?? window.location.pathname;
+    const search = _location?.search ?? window.location.search;
```

Comment:
should we be throwing a better error here if window isn't defined and `_location` isn't passed?

Reply:
Right I think it is nicer.